ludwig/schema/hyperopt/search_algorithm.py

```diff
@@ -54,3 +81,675 @@ def _jsonschema_type_mapping(self):
         },
         default_factory=load_default,
     )
+
+
+@DeveloperAPI
+@register_search_algorithm("random", random_state_field="random_state")
+@register_search_algorithm("variant_generator")
+@ludwig_dataclass
+class BasicVariantSAConfig(BaseSearchAlgorithmConfig):
```

Comment:
Interesting that BasicVariant doesn't require `mode` and `metric` 🤔

Reply:
I'm actually considering removing `mode` and `metric` from the configs since we overwrite those anyway when passing config to `create_searcher`. [Here's where that happens in the original executor code.](https://github.com/ludwig-ai/ludwig/blob/397e46b44a2e39fc8f6c81b94b398c491ff671c0/ludwig/hyperopt/execution.py#L740-L758). WDYT?